banzai_nres/traces.py

```diff
@@ -0,0 +1,183 @@
+"""
+traces.py: Driver scripts for finding echelle orders across a CCD.
+
+Authors
+    G. Mirek Brandt (gmbrandt@ucsb.edu)
+"""
+
+from banzai_nres.utils.trace_utils import fit_traces_order_by_order, Trace
+from banzai_nres.utils import trace_utils
+from banzai_nres.images import NRESImage
+from banzai.stages import Stage
+from banzai.calibrations import CalibrationMaker
+from banzai.utils import fits_utils
+from banzai import dbs
+from banzai.images import DataTable, regenerate_data_table_from_fits_hdu_list
+from astropy.io import fits
+import numpy as np
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class SaveTrace(CalibrationMaker):
+    """
+    Updates the master calibration trace file.
+
+    Note:  if self.try_combinations_of_images, cross_correlate_num is the
+    number of images you compare to see if their trace fits agree
+    """
+    def __init__(self, pipeline_context):
+        super(SaveTrace, self).__init__(pipeline_context)
+        self.pipeline_context = pipeline_context
+        self.master_selection_criteria = self.pipeline_context.CALIBRATION_SET_CRITERIA.get(
+            self.calibration_type.upper(), [])
+        self.fiber_order_header_name = 'FIBRORDR'
+
+    @property
+    def calibration_type(self):
+        return 'TRACE'
+
+    def make_master_calibration_frame(self, images):
+        """
+        :param images: list of Banzai-NRES Image objects.
+        :return: frame with trace coefficients appended as astropy tables etc.
+        """
+        good_frame = images[0]
+
+        make_calibration_name = self.pipeline_context.CALIBRATION_FILENAME_FUNCTIONS[self.calibration_type]
+        master_trace_filename = make_calibration_name(good_frame)
+
+        logger.debug('master_trace', os.path.basename(master_trace_filename))
+
+        header = fits_utils.create_master_calibration_header(images)
+        header[self.fiber_order_header_name] = good_frame.trace.fiber_order
```

Comment:
Why do we need to add the DAY-OBS here? and DATE-OBS, don't those get added automatically?